### PR TITLE
Make start_console thread-safe and remove unnecessary locking for already thread-safe ops

### LINKED ISSUE #5365: pydeephaven's ConsoleService has a racy issue in the start_console() method
It is important that we give all the other methods a careful audit as well.

### PR DESCRIPTION
**Pull request by jmao-denver**
Fixes #5365

### CODE REVIEW

File: py/client/pydeephaven/session.py [hunk 1]
```diff
@@ -384,9 +384,10 @@ def run_script(self, script: str) -> None:
         Raises:
             DHError
         """
-        response = self.console_service.run_script(script)
-        if response.error_message != '':
-            raise DHError("could not run script: " + response.error_message)
+        with self._r_lock:
```
Comment (chipkent) [inline py/client/pydeephaven/session.py:387]: I'm a bit confused about the current design.
1. `_console_service.start_console` does the relevant locking in `_console_services`.
2. If `_console_service.run_script` requires locking, what is that locking not in `_console_services`?
This decision just feels inconsistent.

Comment (chipkent) [inline py/client/pydeephaven/session.py:387]: Why does running a script require locking?

File: py/client/pydeephaven/_console_service.py
Comment (chipkent) [inline py/client/pydeephaven/_console_service.py:1]: The ticket 5465 listed in the description does not exist.

File: py/client/pydeephaven/_console_service.py [hunk 2]
```diff
@@ -19,14 +19,16 @@ def start_console(self):
...
 
-        try:
-            result_id = self.session.make_ticket()
-            response = self._grpc_console_stub.StartConsole(
-                console_pb2.StartConsoleRequest(result_id=result_id, session_type=self.session._session_type),
-                metadata=self.session.grpc_metadata)
-            self.console_id = response.result_id
-        except Exception as e:
-            raise DHError("failed to start a console.") from e
+        with self.session._r_lock:
```
Comment (rcaudy) [inline py/client/pydeephaven/_console_service.py:22]: Is double-checked locking safe in Python?

File: py/client/pydeephaven/session.py (hunk 1 above)
Comment (rcaudy) [inline py/client/pydeephaven/session.py:387]: Why do I need to lock the session when I call run_script?

File: py/client/pydeephaven/_console_service.py (hunk 2 above)
Comment (jmao-denver) [reply py/client/pydeephaven/_console_service.py:22]: Yes, accessing the value of a variable in Python is atomic.

**Review by chipkent — APPROVED** [review]

**Review by rcaudy — APPROVED** [review]

**Pull request merged**